R/datasets.R

```diff
@@ -0,0 +1,641 @@
+#'
+#' @return Converted DataType object.
+#' @export
+#' @md
+data_type_bool <- function() {
+  azureml$data$dataset_factory$DataType$to_bool()
+}
+
+#' Configure conversion to datetime.
+#'
+#' @description
+#' Configure conversion to datetime.
+#'
+#' @param formats Formats to try for datetime conversion. For example `%d-%m-%Y` for data in "day-month-year",
+#' and `%Y-%m-%dT%H:%M:%S.%f` for "combined date an time representation" according to ISO 8601.
+#' * %Y: Year with 4 digits
+#' * %y: Year with 2 digits
+#' * %m: Month in digits
+#' * %b: Month represented by its abbreviated name in 3 letters, like Aug
+#' * %B: Month represented by its full name, like August
+#' * %d: Day in digits
+#' * %H: Hour as represented in 24-hour clock time
+#' * %I: Hour as represented in 12-hour clock time
+#' * %M: Minute in 2 digits
+#' * %S: Second in 2 digits
+#' * %f: Microsecond
+#' * %p: AM/PM designator
+#' * %z: Timezone, for example: -0700
+#'
+#' Format specifiers will be inferred if not specified.
+#' Inference requires that the data source is accessible from current compute.
+#' @return Converted DataType object.
+#' @export
+#' @md
+data_type_datetime <- function(formats = NULL) {
+  azureml$data$dataset_factory$DataType$to_datetime(formats)
+}
+
+#' Configure conversion to 53-bit double.
+#'
+#' @description
+#' Configure conversion to 53-bit double.
+#'
+#' @return Converted DataType object.
+#' @export
+#' @md
+data_type_double <- function()	{
+  azureml$data$dataset_factory$DataType$to_float()
+}
+
+#' Configure conversion to 64-bit integer.
+#'
+#' @description
+#' Configure conversion to 64-bit integer.
+#'
+#' @return Converted DataType object.
+#' @export
+#' @md
+data_type_long <- function() {
+  azureml$data$dataset_factory$DataType$to_float()
+}
+
+#' Configure conversion to string.
+#'
+#' @description
+#' Configure conversion to string.
+#'
+#' @return Converted DataType object.
+#' @export
+#' @md
+data_type_string <- function() {
+  azureml$data$dataset_factory$DataType$to_string()
+}
+
+#' Defines options for how column headers are processed when reading data from files to create a dataset.
+#'
+#' @description
+#' Defines options for how column headers are processed when reading data from files to create a dataset.
+#' These enumeration values are used in the Dataset class method.
+#'
+#' @param option An integer corresponding to an option for how column headers are to be processed
+#' * 0: NO_HEADERS No column headers are read
+#' * 1: ONLY_FIRST_FILE_HAS_HEADERS Read headers only from first row of first file, everything else is data.
+#' * 2: COMBINE_ALL_FILES_HEADERS Read headers from first row of each file, combining identically named columns.
+#' * 3: ALL_FILES_HAVE_SAME_HEADERS Read headers from first row of first file, drops first row from other files.
+#' @return The PromoteHeadersBehavior object.
+#' @export
+#' @md
+promote_headers_behavior <- function(option) {
+  option <- as.integer(option)
```

Comment:
do we need to convert to integer again here?

Reply:
Yes - otherwise Python won't recognize the enum because R defaults to doubles.